app/scripts/controllers/transactions/index.js

```diff
@@ -2106,8 +2113,47 @@ export default class TransactionController extends EventEmitter {
     } else if (contractInteractionTypes) {
       transactionType = TRANSACTION_TYPES.CONTRACT_INTERACTION;
       transactionContractMethod = transactions[id]?.contractMethodName;
+      if (
+        transactionContractMethod === 'Approve' &&
+        tokenStandard === TOKEN_STANDARDS.ERC20
+      ) {
+        if (dappProposedTokenAmount === '0' || customTokenAmount === '0') {
+          transactionApprovalAmountType =
+            TRANSACTION_APPROVAL_AMOUNT_TYPE.REVOKE;
+        } else if (customTokenAmount) {
+          transactionApprovalAmountType =
+            TRANSACTION_APPROVAL_AMOUNT_TYPE.CUSTOM;
+        } else if (dappProposedTokenAmount) {
+          transactionApprovalAmountType =
+            TRANSACTION_APPROVAL_AMOUNT_TYPE.DAPP_PROPOSED;
+        }
+      }
     }
 
+    const transactionApprovalAmountVsProposedRation =
```

Comment:
Can you create helper methods for these and then only call them in the above if block?

Reply:
@danjm Done [here](https://github.com/MetaMask/metamask-extension/pull/15289/commits/cbebce7386de9fa462921da06c1af843b7e38939).